src/sidebar/services/thumbnail.ts

```diff
@@ -9,12 +9,38 @@ import type { FrameSyncService } from './frame-sync';
 export class ThumbnailService {
   /** Map of annotation tag to thumbnail image. */
   #cache = new Map<string, ImageBitmap>();
+
+  /** Max thumbnails to keep in cache. */
+  #cacheSize = 30;
```

Comment:
This number is arbitrary, though it should be good for at least a few screenfuls of annotation cards.